diff --git a/productivity/qzeitgeist/Makefile b/productivity/qzeitgeist/Makefile
index 76b09fb..e846830 100644
--- a/productivity/qzeitgeist/Makefile
+++ b/productivity/qzeitgeist/Makefile
@@ -1,10 +1,10 @@
 # $OpenBSD$
-# Slightly based on FreeBSD port
 
 COMMENT =		Qt wrapper for Zeitgeist, event logging framework
 
-DISTNAME =		QtZeitgeist-0.7.0
-REVISION =		0
+V =			0.8.0
+DISTNAME =		libqzeitgeist-${V}
+PKGNAME =		QtZeitgeist-${V}
 
 SHARED_LIBS =		qzeitgeist	0.0
 
@@ -18,16 +18,22 @@ PERMIT_PACKAGE_FTP =	Yes
 PERMIT_DISTFILES_CDROM =Yes
 PERMIT_DISTFILES_FTP =	Yes
 
-WANTLIB =		lib/qt4/QtDBus lib/qt4/QtTest lib/qt4/QtXml m stdc++
+WANTLIB =		m stdc++
+WANTLIB +=		lib/qt4/QtDBus lib/qt4/QtDeclarative lib/qt4/QtGui
+WANTLIB +=		lib/qt4/QtNetwork lib/qt4/QtTest lib/qt4/QtScript 
+WANTLIB +=		lib/qt4/QtSql lib/qt4/QtSvg lib/qt4/QtXml
+WANTLIB +=		lib/qt4/QtXmlPatterns ${MODPY_WANTLIB}
 
-MASTER_SITES =		http://releases.zeitgeist-project.com/qzeitgeist/
+MASTER_SITES =		${MASTER_SITE_KDE:=stable/libqzeitgeist/${V}/src/}
 EXTRACT_SUFX =		.tar.bz2
 
-MODULES =		devel/cmake x11/qt4
+MODULES =		devel/cmake x11/qt4 lang/python
 BUILD_DEPENDS =		productivity/zeitgeist
 RUN_DEPENDS =		productivity/zeitgeist
 SEPARATE_BUILD =	simple
-WRKDIST =		${WRKDIR}/${DISTNAME}-Source
 NO_REGRESS =		Yes
 
+post-extract:
+	@perl -pi -e 's,/usr/bin/env python,${MODPY_BIN},' \
+		${WRKSRC}/scripts/onto2cpp.py
 .include <bsd.port.mk>
diff --git a/productivity/qzeitgeist/distinfo b/productivity/qzeitgeist/distinfo
index 3cbc757..791b488 100644
--- a/productivity/qzeitgeist/distinfo
+++ b/productivity/qzeitgeist/distinfo
@@ -1,5 +1,5 @@
-MD5 (QtZeitgeist-0.7.0.tar.bz2) = 5Y1LmIu6uBL4vWh42HwJVg==
-RMD160 (QtZeitgeist-0.7.0.tar.bz2) = vO6O6xRDyGDUekTNBb5BbagOyMw=
-SHA1 (QtZeitgeist-0.7.0.tar.bz2) = zWf82QiT3A6SjHNRNDp16X0rCg0=
-SHA256 (QtZeitgeist-0.7.0.tar.bz2) = EsMHMZa6XqhxdZ55vAoez1o6IPHZCr2BTSVV/yEXXX4=
-SIZE (QtZeitgeist-0.7.0.tar.bz2) = 31260
+MD5 (libqzeitgeist-0.8.0.tar.bz2) = l73qahhl231fKck+OkkvJA==
+RMD160 (libqzeitgeist-0.8.0.tar.bz2) = Qtfc8/bfEe3JMsufIc6rdWntx28=
+SHA1 (libqzeitgeist-0.8.0.tar.bz2) = URO+7RgmB2l6esGTqaYxHnND6qM=
+SHA256 (libqzeitgeist-0.8.0.tar.bz2) = CoqpgNZFSczpNpFwWAdoH9fj4Hmkiu5o/EsmU/F9Ya0=
+SIZE (libqzeitgeist-0.8.0.tar.bz2) = 29459
