diff --git a/multimedia/phonon/Makefile b/multimedia/phonon/Makefile
index bd77d6d..4f5886a 100644
--- a/multimedia/phonon/Makefile
+++ b/multimedia/phonon/Makefile
@@ -2,7 +2,7 @@
 
 COMMENT =	multimedia layer for Qt4/KDE4
 CATEGORIES =	multimedia devel x11/kde4
-VERSION =	4.5.1
+VERSION =	4.6.0
 DISTNAME =	phonon-${VERSION}
 
 # Higher than one in Qt4 to avoid issues
@@ -25,7 +25,9 @@ MODULES =	devel/cmake \
 		x11/qt4
 
 WANTLIB =	lib/qt4/QtDBus lib/qt4/QtGui lib/qt4/QtTest lib/qt4/QtXml
-WANTLIB +=	glib-2.0 m qzeitgeist stdc++ pulse pulse-mainloop-glib
+WANTLIB +=	lib/qt4/QtDeclarative lib/qt4/QtNetwork lib/qt4/QtScript
+WANTLIB +=	lib/qt4/QtSql lib/qt4/QtSvg lib/qt4/QtXmlPatterns
+WANTLIB +=	m qzeitgeist stdc++ pulse pulse-mainloop-glib
 
 BUILD_DEPENDS =	x11/kde4/automoc
 
diff --git a/multimedia/phonon/distinfo b/multimedia/phonon/distinfo
index 42d5153..0508630 100644
--- a/multimedia/phonon/distinfo
+++ b/multimedia/phonon/distinfo
@@ -1,5 +1,5 @@
-MD5 (phonon-4.5.1.tar.xz) = PfBxt/mZavzWexiViVXs3Q==
-RMD160 (phonon-4.5.1.tar.xz) = taqTE7YrrH1SM3FttSySdRaO4ZE=
-SHA1 (phonon-4.5.1.tar.xz) = cQqf//5+VY9NDOXqXBGMskj7jag=
-SHA256 (phonon-4.5.1.tar.xz) = AYDwHLbK6QcoWt925K7Om+M69qNS3AV3CeQLeDji9Mg=
-SIZE (phonon-4.5.1.tar.xz) = 291088
+MD5 (phonon-4.6.0.tar.xz) = u+DBxi7RTDFHnEwabPHhcw==
+RMD160 (phonon-4.6.0.tar.xz) = S2A3dExRlhPw45BlqS1nQjBeE2Y=
+SHA1 (phonon-4.6.0.tar.xz) = 2NvBiLWMbdnGpz03QqJSkeZHu5U=
+SHA256 (phonon-4.6.0.tar.xz) = KRXno3ySoKgje55tLvZ7qLAF7jUp0DmRzT0TfwObo8Q=
+SIZE (phonon-4.6.0.tar.xz) = 280712
diff --git a/multimedia/phonon/patches/patch-cmake_CMakeLists_txt b/multimedia/phonon/patches/patch-cmake_CMakeLists_txt
index 3a37c35..54cf9a6 100644
--- a/multimedia/phonon/patches/patch-cmake_CMakeLists_txt
+++ b/multimedia/phonon/patches/patch-cmake_CMakeLists_txt
@@ -1,17 +1,18 @@
 $OpenBSD$
---- cmake/CMakeLists.txt.orig	Tue Jan 10 18:17:08 2012
-+++ cmake/CMakeLists.txt	Tue Jan 10 18:17:34 2012
-@@ -1,13 +1,9 @@
- install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in
-               ${CMAKE_CURRENT_SOURCE_DIR}/COPYING-CMAKE-SCRIPTS
--              ${CMAKE_CURRENT_SOURCE_DIR}/FindAutomoc4.cmake
--              ${CMAKE_CURRENT_SOURCE_DIR}/FindPackageHandleStandardArgs.cmake
-               ${CMAKE_CURRENT_SOURCE_DIR}/FindPhononInternal.cmake
-               ${CMAKE_CURRENT_SOURCE_DIR}/PhononMacros.cmake
--              ${CMAKE_CURRENT_SOURCE_DIR}/FindQt4.cmake
-               ${CMAKE_CURRENT_SOURCE_DIR}/MacroEnsureVersion.cmake
-               ${CMAKE_CURRENT_SOURCE_DIR}/MacroLogFeature.cmake
-               ${CMAKE_CURRENT_SOURCE_DIR}/MacroOptionalFindPackage.cmake
--              ${CMAKE_CURRENT_SOURCE_DIR}/MacroPushRequiredVars.cmake
-               ${CMAKE_CURRENT_SOURCE_DIR}/PhononMacros.cmake
-         DESTINATION ${BUILDSYSTEM_INSTALL_DIR})
+--- cmake/CMakeLists.txt.orig	Sat Jan  7 15:43:25 2012
++++ cmake/CMakeLists.txt	Sat Jan  7 16:33:27 2012
+@@ -1,14 +1,10 @@
+ install(FILES cmake_uninstall.cmake.in
+               COPYING-CMAKE-SCRIPTS
+-              FindAutomoc4.cmake
+-              FindPackageHandleStandardArgs.cmake
+               FindPhononInternal.cmake
+               PhononMacros.cmake
+-              FindQt4.cmake
+               MacroEnsureVersion.cmake
+               MacroLogFeature.cmake
+               MacroOptionalFindPackage.cmake
+-              MacroPushRequiredVars.cmake
+               PhononMacros.cmake
+               Qt4ConfigDependentSettings.cmake
+               Qt4Macros.cmake
diff --git a/multimedia/phonon/phonon.port.mk b/multimedia/phonon/phonon.port.mk
index ba1a2f9..316ea5b 100644
--- a/multimedia/phonon/phonon.port.mk
+++ b/multimedia/phonon/phonon.port.mk
@@ -3,7 +3,7 @@
 MODPHONON_LIB_VERSION =	6
 
 MODPHONON_WANTLIB =	phonon=${MODPHONON_LIB_VERSION}
-MODPHONON_LIB_DEPENDS =	phonon->=4.5.1:multimedia/phonon
+MODPHONON_LIB_DEPENDS =	phonon->=4.6.0:multimedia/phonon
 MODPHONON_RUN_DEPENDS =	phonon-gstreamer-*|phonon-vlc-*:multimedia/phonon-backend/gstreamer
 
 MODULES +=	x11/qt4
