Index: Makefile
===================================================================
RCS file: /home/amit/all/openbsd_cvsync/ports/devel/py-tz/Makefile,v
retrieving revision 1.27
diff -u -p -u -p -r1.27 Makefile
--- Makefile	5 Sep 2013 08:34:47 -0000	1.27
+++ Makefile	25 Nov 2013 00:01:30 -0000
@@ -2,11 +2,10 @@
 
 COMMENT =		Python API for dealing with timezones
 
-MODPY_EGG_VERSION =	2013b
+MODPY_EGG_VERSION =	2013.8
 DISTNAME =		pytz-${MODPY_EGG_VERSION}
 PKGNAME =		${DISTNAME:S/^py/py-/}
 CATEGORIES =		devel
-REVISION =		0
 
 HOMEPAGE =		http://pytz.sourceforge.net/
 MAINTAINER =		frantisek holop <minusf@obiit.org>
Index: distinfo
===================================================================
RCS file: /home/amit/all/openbsd_cvsync/ports/devel/py-tz/distinfo,v
retrieving revision 1.13
diff -u -p -u -p -r1.13 distinfo
--- distinfo	3 May 2013 10:12:10 -0000	1.13
+++ distinfo	25 Nov 2013 00:01:44 -0000
@@ -1,2 +1,2 @@
-SHA256 (pytz-2013b.tar.gz) = B5cpPRLKL35d1bwIB/Uh851bYKLDR8SPg2L5/QF0rz8=
-SIZE (pytz-2013b.tar.gz) = 331277
+SHA256 (pytz-2013.8.tar.gz) = WFUuhwqiwKH6O075I/APvz5Vr6qH+NMSRNRPGI3keTo=
+SIZE (pytz-2013.8.tar.gz) = 297061
Index: pkg/PLIST
===================================================================
RCS file: /home/amit/all/openbsd_cvsync/ports/devel/py-tz/pkg/PLIST,v
retrieving revision 1.14
diff -u -p -u -p -r1.14 PLIST
--- pkg/PLIST	5 Sep 2013 19:05:42 -0000	1.14
+++ pkg/PLIST	25 Nov 2013 00:02:02 -0000
@@ -7,16 +7,17 @@ lib/python${MODPY_VERSION}/site-packages
 lib/python${MODPY_VERSION}/site-packages/pytz-${MODPY_EGG_VERSION}-py${MODPY_VERSION}.egg-info/top_level.txt
 lib/python${MODPY_VERSION}/site-packages/pytz-${MODPY_EGG_VERSION}-py${MODPY_VERSION}.egg-info/zip-safe
 lib/python${MODPY_VERSION}/site-packages/pytz/__init__.py
-${MODPY_COMMENT}lib/python${MODPY_VERSION}/site-packages/pytz/${MODPY_PYCACHE}/
-lib/python${MODPY_VERSION}/site-packages/pytz/${MODPY_PYCACHE}__init__.${MODPY_PYC_MAGIC_TAG}pyc
-lib/python${MODPY_VERSION}/site-packages/pytz/${MODPY_PYCACHE}exceptions.${MODPY_PYC_MAGIC_TAG}pyc
-lib/python${MODPY_VERSION}/site-packages/pytz/${MODPY_PYCACHE}reference.${MODPY_PYC_MAGIC_TAG}pyc
-lib/python${MODPY_VERSION}/site-packages/pytz/${MODPY_PYCACHE}tzfile.${MODPY_PYC_MAGIC_TAG}pyc
-lib/python${MODPY_VERSION}/site-packages/pytz/${MODPY_PYCACHE}tzinfo.${MODPY_PYC_MAGIC_TAG}pyc
+lib/python${MODPY_VERSION}/site-packages/pytz/__init__.pyc
 lib/python${MODPY_VERSION}/site-packages/pytz/exceptions.py
+lib/python${MODPY_VERSION}/site-packages/pytz/exceptions.pyc
+lib/python${MODPY_VERSION}/site-packages/pytz/lazy.py
+lib/python${MODPY_VERSION}/site-packages/pytz/lazy.pyc
 lib/python${MODPY_VERSION}/site-packages/pytz/reference.py
+lib/python${MODPY_VERSION}/site-packages/pytz/reference.pyc
 lib/python${MODPY_VERSION}/site-packages/pytz/tzfile.py
+lib/python${MODPY_VERSION}/site-packages/pytz/tzfile.pyc
 lib/python${MODPY_VERSION}/site-packages/pytz/tzinfo.py
+lib/python${MODPY_VERSION}/site-packages/pytz/tzinfo.pyc
 lib/python${MODPY_VERSION}/site-packages/pytz/zoneinfo/
 lib/python${MODPY_VERSION}/site-packages/pytz/zoneinfo/Africa/
 lib/python${MODPY_VERSION}/site-packages/pytz/zoneinfo/Africa/Abidjan
