Index: Makefile
===================================================================
RCS file: /cvs/ports/x11/openmotif/Makefile,v
retrieving revision 1.56
diff -u -p -r1.56 Makefile
--- Makefile	29 Oct 2012 12:16:48 -0000	1.56
+++ Makefile	18 Nov 2012 17:13:26 -0000
@@ -4,7 +4,7 @@ COMMENT-main=		Motif toolkit
 COMMENT-demos=		Motif toolkit demos
 
 VERSION=		2.3.4
-REVISION-main=		0
+REVISION-main=		1
 DISTNAME=		motif-${VERSION}
 EXTRACT_SUFX=		-src.tgz
 PKGNAME-main=		openmotif-${VERSION}
@@ -20,6 +20,8 @@ DIST_SUBDIR=		openmotif
 
 HOMEPAGE=		http://motif.ics.com/
 
+MAINTAINER=		Pascal Stumpf <Pascal.Stumpf@cubes.de>
+
 # LGPLv2.1
 PERMIT_PACKAGE_CDROM=	Yes
 PERMIT_DISTFILES_CDROM=	Yes
@@ -35,10 +37,10 @@ AUTOMAKE_VERSION=	1.12
 CONFIGURE_ARGS=		--enable-themes \
 			--enable-xft \
 			--enable-jpeg \
-			--disable-printing \
+			--enable-printing \
 			--enable-png
 CONFIGURE_ENV=		CPPFLAGS="-I${X11BASE}/include -I${LOCALBASE}/include -I${LOCALBASE}/include/libpng" \
-			LDFLAGS="-L${LOCALBASE}/lib -lm"
+			LDFLAGS="-L${X11BASE}/lib -L${LOCALBASE}/lib -lm"
 MAKE_ENV=		AUTOCONF_VERSION=${AUTOCONF_VERSION} \
 			AUTOMAKE_VERSION=${AUTOMAKE_VERSION}
 
@@ -53,16 +55,18 @@ MODULES=		converters/libiconv
 
 BUILD_DEPENDS=		${MODGNU_AUTOCONF_DEPENDS} \
 			${MODGNU_AUTOMAKE_DEPENDS} \
-			devel/libtool
+			devel/libtool \
+			print/xprintproto
 
 LIB_DEPENDS=		graphics/jpeg \
-			graphics/png
+			graphics/png \
+			print/libXp
 LIB_DEPENDS-demos=	x11/openmotif \
 			${MODLIBICONV_LIB_DEPENDS} \
 			${LIB_DEPENDS}
-WANTLIB=		ICE SM X11 Xext Xt c Xft Xmu Xrender \
+WANTLIB=		ICE SM X11 Xext Xt c Xau Xft Xmu Xrender \
 			expat fontconfig freetype m z xcb \
-			jpeg png
+			jpeg png Xp
 WANTLIB-demos=		${WANTLIB} Xm Mrm
 
 # XXX  Large, complicated, only partially automated test suite available
