From 7a3be2b28d6d4f41831be6ffbd94245ce962a3e9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Marie?= <semarie@users.noreply.github.com>
Date: Wed, 7 Jan 2015 11:52:28 +0100
Subject: [PATCH] librustdoc: openbsd support

---
 src/librustdoc/flock.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/librustdoc/flock.rs b/src/librustdoc/flock.rs
index dcc9011..a5ae6ea 100644
--- a/src/librustdoc/flock.rs
+++ b/src/librustdoc/flock.rs
@@ -63,7 +63,7 @@ mod imp {
         pub const F_SETLKW: libc::c_int = 13;
     }
 
-    #[cfg(target_os = "dragonfly")]
+    #[cfg(any(target_os = "dragonfly", target_os = "openbsd"))]
     mod os {
         use libc;
 
-- 
2.2.2

