From 35fb4c5eaa2db928420bda53e4eed577cae8db8f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Marie?= <semarie@users.noreply.github.com>
Date: Wed, 7 Jan 2015 11:53:11 +0100
Subject: [PATCH 04/22] libbacktrace: openbsd support

---
 src/libbacktrace/configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libbacktrace/configure b/src/libbacktrace/configure
index f4f9d05..14a9fd4 100755
--- a/src/libbacktrace/configure
+++ b/src/libbacktrace/configure
@@ -9332,7 +9332,7 @@ if test -z "$aix_libpath"; then aix_libpath="/usr/lib:/lib"; fi
       ;;
 
     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
-    freebsd* | dragonfly*)
+    freebsd* | dragonfly* | openbsd*)
       archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
       hardcode_libdir_flag_spec='-R$libdir'
       hardcode_direct=yes
-- 
2.2.2

