From d064bf44d877be88c8e4325a2adffd211e4871c4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Marie?= <semarie@users.noreply.github.com>
Date: Sun, 11 Jan 2015 12:35:31 +0100
Subject: [PATCH] liblibc: complete definition of struct stat

---
 src/liblibc/lib.rs | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/liblibc/lib.rs b/src/liblibc/lib.rs
index b992dac..562a9f8 100644
--- a/src/liblibc/lib.rs
+++ b/src/liblibc/lib.rs
@@ -1525,7 +1525,6 @@ pub mod types {
                     pub st_dev: dev_t,
                     pub st_ino: ino_t,
                     pub st_nlink: nlink_t,
-                    //pub st_padding1: uint16_t,
                     pub st_uid: uid_t,
                     pub st_gid: gid_t,
                     pub st_rdev: dev_t,
@@ -1540,9 +1539,8 @@ pub mod types {
                     pub st_blksize: blksize_t,
                     pub st_flags: fflags_t,
                     pub st_gen: uint32_t,
-                    //pub st_lspare: int32_t,
-                    //pub st_qspare1: int64_t,
-                    //pub st_qspare2: int64_t,
+                    pub st_birthtime: time_t,
+                    pub st_birthtime_nsec: c_long,
                 }
                 #[repr(C)]
                 #[derive(Copy)] pub struct utimbuf {
-- 
2.2.2

