From 48ae17c97ac7043e23ccae294cf0dbd7a85746ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Marie?= <semarie@users.noreply.github.com>
Date: Tue, 13 Jan 2015 11:09:08 +0100
Subject: [PATCH 10/22] libstd: openbsd provide clock_gettime via libc

---
 src/libstd/time/mod.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libstd/time/mod.rs b/src/libstd/time/mod.rs
index d6c94f2..9ae9458 100644
--- a/src/libstd/time/mod.rs
+++ b/src/libstd/time/mod.rs
@@ -68,7 +68,8 @@ mod imp {
     use libc::{c_int, timespec};
 
     // Apparently android provides this in some other library?
-    #[cfg(not(target_os = "android"))]
+    // OpenBSD provide it via libc
+    #[cfg(not(any(target_os = "android", target_os = "openbsd")))]
     #[link(name = "rt")]
     extern {}
 
-- 
2.2.2

