From 50580cc532970aaa093a301412e4306adfe4752c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Marie?= <semarie@users.noreply.github.com>
Date: Tue, 13 Jan 2015 11:23:06 +0100
Subject: [PATCH 12/22] libstd: dl* come from libc under openbsd

---
 src/libstd/dynamic_lib.rs | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/libstd/dynamic_lib.rs b/src/libstd/dynamic_lib.rs
index bb97cee..856ef11 100644
--- a/src/libstd/dynamic_lib.rs
+++ b/src/libstd/dynamic_lib.rs
@@ -259,6 +259,7 @@ pub mod dl {
         Local = 0,
     }
 
+    #[cfg(not(target_os = "openbsd"))]
     #[link_name = "dl"]
     extern {
         fn dlopen(filename: *const libc::c_char,
@@ -268,6 +269,16 @@ pub mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
+    extern {
+        fn dlopen(filename: *const libc::c_char,
+                  flag: libc::c_int) -> *mut libc::c_void;
+        fn dlerror() -> *mut libc::c_char;
+        fn dlsym(handle: *mut libc::c_void,
+                 symbol: *const libc::c_char) -> *mut libc::c_void;
+        fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
+    }
 }
 
 #[cfg(target_os = "windows")]
-- 
2.2.2

